lib/arel/visitors/sqlserver.rb

```diff
@@ -63,7 +68,36 @@ def visit_Arel_Nodes_Grouping(o, collector)
         super
       end
 
-      def visit_Arel_Nodes_SelectStatement o, collector
+      def visit_Arel_Nodes_HomogeneousIn(o, collector)
+        collector.preparable = false
+
+        collector << quote_table_name(o.table_name) << "." << quote_column_name(o.column_name)
+
+        if o.type == :in
+          collector << " IN ("
+        else
+          collector << " NOT IN ("
+        end
+
+        values = o.casted_values
+
+        if values.empty?
+          collector << @connection.quote(nil)
+        else
+          # Monkey-patch start. Add query attribute bindings rather than just values.
```

Comment:
Does that mean that Rails 6.1 is not parameterizing queries with `IN` clauses anymore?

Reply:
Rails 6.1 is still parameterizing queries with IN clauses. The difference between PostgreSQL/MySQL/SQLite and SQL Server is that we require the type information for the query.

```ruby
Topic.where(id: [1, 2])
```

PostgreSQL
> SELECT "topics".* FROM "topics" WHERE "topics"."id" IN ($1, $2)

MySQL
> SELECT `topics`.* FROM `topics` WHERE `topics`.`id` IN (1, 2)

SQLite
> SELECT "topics".* FROM "topics" WHERE "topics"."id" IN (?, ?)

SQL Server
> EXEC sp_executesql N'SELECT [topics].* FROM [topics] WHERE [topics].[id] IN (@0, @1)', N'@0 bigint, @1 bigint', @0 = 1, @1 = 2